op-batcher/batcher/channel_manager_test.go

```diff
@@ -478,3 +485,70 @@ func ChannelManagerCloseAllTxsFailed(t *testing.T, batchType uint) {
 	_, err = m.TxData(eth.BlockID{})
 	require.ErrorIs(err, io.EOF, "Expected closed channel manager to produce no more tx data")
 }
+
+func TestChannelManager_ChannelCreation(t *testing.T) {
+	l := testlog.Logger(t, log.LevelCrit)
+	maxChannelDuration := uint64(15)
+	cfg := ChannelConfig{
+		MaxChannelDuration: maxChannelDuration,
+		MaxFrameSize:       1000,
+		CompressorConfig: compressor.Config{
+			TargetNumFrames:  100,
+			TargetFrameSize:  1000,
+			ApproxComprRatio: 1.0,
+		},
+	}
+
+	for _, tt := range []struct {
+		name                   string
+		safeL1Block            eth.BlockID
+		lastL1Block            eth.BlockID
+		expectedChannelTimeout uint64
+	}{
+		{
+			name: "UseSafeHeadWhenNoLastL1Block",
+			safeL1Block: eth.BlockID{
+				Number: uint64(123),
+			},
+			lastL1Block: eth.BlockID{},
+			// Safe head + maxChannelDuration
+			expectedChannelTimeout: 123 + maxChannelDuration,
+		},
+		{
+			name: "UseLastL1Block",
+			safeL1Block: eth.BlockID{
+				Number: uint64(123),
+			},
+			lastL1Block: eth.BlockID{
+				Number: uint64(456),
+			},
+			// LastL1Block + maxChannelDuration
+			expectedChannelTimeout: 456 + maxChannelDuration,
+		},
+		{
+			name:        "NoLastL1BlockNoSafeL1Block",
+			safeL1Block: eth.BlockID{},
+			lastL1Block: eth.BlockID{},
+			// No timeout
+			expectedChannelTimeout: 0,
+		},
+	} {
+		test := tt
+		t.Run(fmt.Sprintf("TestChannelManager_ChannelCreation_%s", test.name), func(t *testing.T) {
```

Comment:
```suggestion
		t.Run(test.name), func(t *testing.T) {
```
the parent test name is already prepended